api/commands/prune.ts

```diff
@@ -77,5 +77,12 @@ function computeIncludedActionNames(
     }
   }
 
+  // Add auto assertions
+  [...compiledGraph.assertions].forEach(assertion => {
+    if (includedActionNames.has(assertion.parent) && !includedActionNames.has(assertion.name)) {
```

Comment:
you can remove `!includedActionNames.has(assertion.name)`, the point in a `Set` is that all this is handled for you